src/pages/en/guides/migrate-to-astro/from-nextjs.mdx

```diff
@@ -2,72 +2,924 @@
+
+2. Replace both these files with a single Astro layout file that contains a page shell (`<html>`, `<head>`, and `<body>` tags) and a `<slot/>` instead of React's `{children}` prop:
+
+```astro title="src/layouts/Layout.astro"
+<html lang="en">
+	<head>
+	    <title>My Page</title>
+	</head>
+	<body>
+		<slot/>
+	</body>
+</html>
+```
+
+### Migrating Pages and Posts
+
+In Next.js, your posts either live in `/pages` or `/app/routeName/page.jsx`.
+
+In Astro, **all your page content must live within `src/`**.
+
+#### React pages
+
+Your existing Next JSX (`.js`) pages will need to be [converted from JSX files to `.astro` pages](#convert-jsx-files-to-astro-files). You cannot use an existing JSX page file in Astro.
+
+These [`.astro` pages](/en/core-concepts/pages/) must be located within `src/pages/` and will have page routes generated automatically based on their file path.
+
+#### Markdown and MDX pages
+
+Astro has built-in support for Markdown and an optional integration for MDX files. You can reuse any existing [Markdown and MDX files](/en/guides/markdown-content/), but they may require some adjustments to their frontmatter, such as adding [Astro's special `layout` frontmatter property](/en/core-concepts/layouts/#markdownmdx-layouts). You will no longer need to manually create pages for each Markdown-generated route. These files can be placed within `src/pages/` to take advantage of automatic file-based routing.
+
+When part of a [content collection](/en/guides/content-collections/), Markdown and MDX files will live in folders within `src/content/` and you will [generate those pages dynamically](/en/guides/content-collections/#generating-pages-from-content-collections).
+
+### Migrating Tests
+
+As Astro outputs raw HTML, it is possible to write end-to-end tests using the output of the build step. Any end-to-end tests written previously might work out-of-the-box if you have been able to match the markup of your Next site. Testing libraries such as Jest and React Testing Library can be imported and used in Astro to test your React components.
+
+See Astro's [testing guide](/en/guides/testing/) for more.
+
+## Reference: Convert Next.js Syntax to Astro
+
+### Next Links to Astro
+
+Convert any Next `<Link to="">`, `<NavLink>` etc. components to HTML `<a href="">` tags. 
+
+```astro del={1} ins={2}
+<Link to="/blog">Blog</Link>
+<a href="/blog">Blog</a>
+```
+
+Astro does not use any special component for links, although you are welcome to build custom link components. You can then import and use this `<Link>` just as you would any other component.
+
+```astro title="src/components/Link.astro"
+---
+const { to } = Astro.props
+---
+<a href={to}><slot /></a>
+```
+
+### Next Imports to Astro
+
+If necessary, update any [file imports](/en/guides/imports/) to reference relative file paths exactly. This can be done using [import aliases](/en/guides/typescript/#import-aliases), or by writing out a relative path in full. 
```

Comment:
Rather than "if necessary", we can spell out when it is necessary (I'm not sure from reading this)